src/components/Clock.tsx

```diff
@@ -0,0 +1,66 @@
+import { Component } from 'react';
+
+function getDate(): string {
+  const value = new Date();
+
+  return value.toUTCString().slice(-12, -4);
+}
+
+type State = {
+  date: string,
+  timer: number,
+};
+
+type Props = {
+  name: string,
+};
+
+export class Clock extends Component<Props, State> {
+  state: Readonly<State> = {
+    date: getDate(),
+    timer: 0,
+  };
+
+  componentDidMount(): void {
+    this.setState({
+      timer: window.setInterval(() => {
+        this.setState({ date: getDate() });
+      }, 1000),
+    });
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
+    }
+
+    if (prevState.date !== this.state.date) {
+      // eslint-disable-next-line no-console
+      console.info(this.state.date);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.state.timer);
+  }
```

Comment:
It is not necessary to put your timer in the state. You can write it just inside class component to avoid overcalls in your code